tests/unit/acquisition/test_function.py

```diff
@@ -461,6 +472,153 @@ def prepare_acquisition_function(
+
+class _PseudoTrainableLinear(GaussianProcess, PseudoTrainableProbModel):
+    def __init__(
+        self,
+        *,
+        kernel_amplitude: float | TensorType | None = None,
+    ):
+        kernel = tfp.math.psd_kernels.ExponentiatedQuadratic(kernel_amplitude)
+        super().__init__([lambda x: tf.reduce_sum(x, axis=-1, keepdims=True)], [kernel])
+
+    def __repr__(self) -> str:
+        return "_PseudoTrainableLinear()"
+
+
+_mo_test_models = (
+    _PseudoTrainableQuadratic,
+    _PseudoTrainableLinear,
+    _PseudoTrainableQuadratic,
+)
+
+
+def test_ehvi_builder_raises_for_empty_data() -> None:
+    num_obj = 3
+    dataset = Dataset(tf.zeros([0, 2]), tf.zeros([0, 1]))
+    model = ModelStack(*[(_PseudoTrainableQuadratic(), 1) for _ in range(num_obj)])
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        ExpectedHypervolumeImprovement().prepare_acquisition_function(dataset, model)
+
+
+def test_ehvi_builder_builds_expected_hv_improvement_using_pareto_from_model() -> None:
+    num_obj = 2
+    train_x = tf.constant([[-2.0], [-1.5], [-1.0], [0.0], [0.5], [1.0], [1.5], [2.0]])
+    dataset = Dataset(
+        train_x,
+        tf.tile(
+            tf.constant([[4.1], [0.9], [1.2], [0.1], [-8.8], [1.1], [2.1], [3.9]]), [1, num_obj]
+        ),
+    )
+
+    model = ModelStack(*[(_mo_test_models[_](), 1) for _ in range(num_obj)])
+    acq_fn = ExpectedHypervolumeImprovement().prepare_acquisition_function(dataset, model)
+
+    model_pred_observation = model.predict(train_x)[0]
+    _prt = Pareto(model_pred_observation)
+    xs = tf.linspace([[-10.0]], [[10.0]], 100)
+    expected = expected_hv_improvement(model, _prt, get_reference_point(_prt.front))(xs)
+    npt.assert_allclose(acq_fn(xs), expected)
+
+
+@pytest.mark.parametrize("at", [tf.constant([[0.0], [1.0]]), tf.constant([[[0.0], [1.0]]])])
+def test_ehvi_raises_for_invalid_batch_size(at: TensorType) -> None:
+    num_obj = 2
+    train_x = tf.constant([[-2.0], [-1.5], [-1.0], [0.0], [0.5], [1.0], [1.5], [2.0]])
+
+    model = ModelStack(*[(_mo_test_models[_](), 1) for _ in range(num_obj)])
+    model_pred_observation = model.predict(train_x)[0]
+    _prt = Pareto(model_pred_observation)
+    ehvi = expected_hv_improvement(model, _prt, get_reference_point(_prt.front))
+
+    with pytest.raises(TF_DEBUGGING_ERROR_TYPES):
+        ehvi(at)
+
+
+@random_seed
+@pytest.mark.parametrize(
+    "input_dim, num_samples_per_point, existing_observations, obj_num, variance_scale ,rtol, atol",
+    [
+        (
+            1,
+            50_000,
+            tf.constant([[0.3, 0.2], [0.2, 0.22], [0.1, 0.25], [0.0, 0.3]]),
+            2,
+            1.0,
+            0.01,
+            1e-2,
+        ),
+        (
+            1,
+            200_000,
+            tf.constant([[0.3, 0.2], [0.2, 0.22], [0.1, 0.25], [0.0, 0.3]]),
+            2,
+            2.0,
+            0.01,
+            1e-2,
+        ),
+        (2, 50_000, tf.constant([[0.0, 0.0]]), 2, 1.0, 0.01, 1e-2),
```

Comment:
Would be nice to also have a test with input_dim 3 or even 4?

(Also for parametrized tests with lots of parameters like here it can often be useful to use [pytest.param](https://docs.pytest.org/en/stable/reference.html#pytest-param) to assign the cases text ids, to make it easier to see what each case is testing.)

Reply:
Added `input_dim` 3 and 4 test case, also add a case with 3 obj as now the partition for obj > 2 is supported by #207.
Test cases with the text ids are also used.